x-pack/metricbeat/module/gcp/metrics/metrics_requester_test.go

```diff
@@ -247,6 +75,83 @@ func TestGetTimeIntervalAligner(t *testing.T) {
 	}
 }
 
+func TestGetFilterForMetric(t *testing.T) {
+	if err := logp.DevelopmentSetup(logp.ToObserverOutput()); err != nil {
+		t.Fatalf("cannot initialise logger on development mode: %+v", err)
+	}
+
+	var logger = logp.NewLogger("TestGetFilterForMetric")
+
+	cases := []struct {
+		title          string
+		s              string
+		m              string
+		r              metricsRequester
+		expectedFilter string
+	}{
+		{
+			"compute service with empty config",
+			gcp.ServiceCompute,
```

Comment:
Here `gcp.ServiceCompute` is used as an example for the overall behaviour. All specific behaviours are tested in isolation in other functions:
- `TestBuildRegionsFilter` verifies that the region based filter creation works as intended
- `TestIsAGlobalService` verifies each service (and default) corresponding "global" status flag value
- `TestGetServiceLabelFor` verifies each service (and default) corresponding label value
- `TestTrimWildcard` verifies the wildcard remover produces correct results
- `TestBuildLocationFilter` verifies the location based filter creation works as intended

Keeping tests `TestIsAGlobalService` and `TestGetServiceLabelFor` up to date with the available services and labels in the `gcp` package will ensure all combinations to be covered.